lib/debug/server_dap.rb

```diff
@@ -621,6 +633,17 @@ def process_dap args
       req = args.shift
 
       case type
+      when :breakpoints
+        bps = req.dig("arguments", "breakpoints")
+        SESSION.clear_method_breakpoints
+
+        breakpoints = bps.map do |bp|
+          if bp["name"]&.match(Session::METHOD_SIGNATURE_REGEXP)
+            make_breakpoint [:method, $1, $2, $3, bp["condition"]]
```

Comment:
`bp["condition"]` here could be the empty string if the user first creates a condition, then deletes the condition. We should treat an empty string like `nil` (i.e. no condition) instead of like a `false` condition, so we should turn the empty string into `nil`.